src/qibojit/backends/gpu.py

```diff
@@ -132,6 +132,8 @@ def cast(self, x, dtype=None, copy=False):
     def to_numpy(self, x):
         if isinstance(x, self.cp.ndarray):
             return x.get()
+        elif isinstance(x, list):
+            return self.cp.asarray(x).get()
```

Comment:
If it's a list it is already on CPU so in principle you could directly do
```suggestion
            return np.asarray(x)
```
unless it is a list of cupy arrays in which case you should keep the current approach because what I propose will fail.

Reply:
unfortunately it is a list of cupy arrays